docs/source/about/contract.rst

```diff
@@ -63,64 +91,86 @@ MUST EITHER:
 ------------------------------
 
-Most services will want to use Jenkins to drive the deploy process. You can think of our Jenkins tooling as a reference implementation of the manual steps (below). Services that want this:
+Most services will want to use Jenkins to drive the deploy process.
+You can think of our Jenkins tooling as a reference implementation of the manual steps (below).
+Services that want this:
+
+* MUST contain a Makefile.
+  This is how Jenkins will manipulate the service.
+
+* The Makefile MUST respond to ``make test``.
+
+  * This makefile target SHOULD run unit tests.
+
+* The Makefile MUST respond to ``make itest``, which:
+
+   * MUST generate a Docker image which will run with the currently checked-out code.
+   * SHOULD respond by running some kind of system-level test (integration, e2e, contract, smoke) which exercises the generated Docker image
+   * MUST tag the generated docker image with the tag specified in the ``DOCKER_TAG`` environment variable.
+     Jenkins will calculate a tag for the newly-built Docker image and pass it to the service’s build scripts via this parameter::
 
-* MUST contain a Makefile. This is how Jenkins will manipulate the service.
-* The Makefile SHOULD delegate non-trivial work to helper scripts written in an appropriate language
-   * Helper scripts MAY live in paasta/ and MAY be named after the Makefile targets that call them, e.g. paasta/test, paasta/itest,  etc.
-* MUST respond to make test
-* SHOULD respond to ``make test`` by running unit tests
-* MUST respond to ``make itest`` by generating a Docker image which will run with the currently checked-out code
-* SHOULD respond by running some kind of system-level test (integration, e2e, contract, smoke) which exercises the generated Docker image
-* MUST accept a parameter $(DOCKER_TAG) to make itest. Jenkins will calculate a tag for the newly-built Docker image and pass it to the service’s build scripts via this parameter::
+       # $(DOCKER_TAG) SHOULD default to something sensible when the job is run outside of Jenkins (e.g. during local development).
+       # E.g.:
+       DOCKER_TAG ?= <servicename>-dev-$(USER)
+       ...
+       build:
+       docker build -t $(DOCKER_TAG) .
 
-    $(DOCKER_TAG) SHOULD default to something sensible when the job is run outside of Jenkins (e.g. during local development).
-    E.g.:
-    DOCKER_TAG ?= <servicename>-dev-$(USER)
-    ...
-    build:
-    docker build -t $(DOCKER_TAG) .
+* The Makefile SHOULD respond to ``make cook-image`` by generating a Docker image tagged with ``$(DOCKER_TAG)`` for use by local-run.
 
-* SHOULD respond to make cook-image by generating a Docker image tagged with $(DOCKER_TAG) for use by local-run
-* cook-image MAY be added as a dependency of the itest target so that both Jenkins and local-run use the same image creation process::
+   * cook-image MAY be added as a dependency of the itest target so that both Jenkins and local-run use the same image creation process::
```

Comment:
I wonder if we should change the `SHOULD` and `MAY` here with `MUST`

supporting cook-image and local-run building an image in different ways seems like something we'd want to avoid (and requiring cook-image would let CID do some nice optimizations in a "guaranteed" way

Reply:
Yeah, we can. That's just a more structural change than I wanted to do in this PR. I'd 🚢  a separate PR to `s/SHOULD|MAY/MUST/` on those two lines.